behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx

```diff
@@ -86,19 +84,243 @@ export function getServices(service_publication, params, filter, requireArtist =
+    });
+
+    // get user profile image from database
+    const isLoadingImages = useSubscribe('profile_images');
+    let imagesData = useTracker(() => {
+        return ImageCollection.find({"imageType": "profile"}).fetch();
+    });
+
+    const isLoading = isLoadingUsers() || isLoadingImages();
+
+    // manual aggregation user data with their image
+    for (let i = 0; i < usersData.length; i++) {
+        for (let j = 0; j < imagesData.length; j++) {
+
+            // find matching image for the artist
+            if (imagesData[j].imageType === "profile" && usersData[i].username === imagesData[j].target_id) {
+                usersData[i].profileImageData = imagesData[j].imageData;
+                break;
+            }
+        }
+    }
+    console.log("usersData", usersData)
+
+    return [isLoading, usersData]
 }
 
+/**
+ * Loads a specific service with its relevant service and user data
+ *
+ * @param serviceId - ID of the service to get data
+ * @returns {[boolean, Object, Object, Object, Object]} - returns a boolean isLoading that is true when loading, false when finished loading.
+ * The objects contain serviceData and serviceImagesData as well as related artistData and their profile image data.
+ */
+export function getSpecificService(serviceId) {
+
+    // will always just get 1 specific service by ID
+    const isLoadingService = useSubscribe('specific_service', serviceId);
+    let serviceData = useTracker(() => {
+        return ServiceCollection.find({
+            _id: serviceId
+        }).fetch()[0];
+    });
+
+    // get service images from database
+    const isLoadingServiceImages = useSubscribe('specific_service_images', serviceId);
+    let serviceImagesData = useTracker(() => {
+        return ImageCollection.find({
+            $and: [
+                {"imageType": "service"},
+                {"target_id": serviceId}
+            ]
+        }).fetch();
+    });
+
+    // get artist data + profile image from database
+    const serviceArtistUsername = serviceData ? serviceData.artistUsername : "";
+    const isLoadingArtist = useSubscribe('specific_user', serviceArtistUsername);
+    let artistData = useTracker(() => {
+        return UserCollection.find({
+            username: serviceArtistUsername
+        }).fetch()[0];
+    });
+
+    const isLoadingArtistProfile = useSubscribe('specific_profile_image', serviceArtistUsername);
+    let profileImageData = useTracker(() => {
+        return ImageCollection.find({
+            $and: [
+                {"imageType": "profile"},
+                {"target_id": serviceArtistUsername}
+            ]
+        }).fetch()[0];
+    });
+
+    const isLoading = isLoadingService() || isLoadingArtist() || isLoadingServiceImages() || isLoadingArtistProfile();
+
+    console.log("service data", [serviceData, artistData, serviceImagesData, profileImageData])
+
+    return [isLoading, serviceData, artistData, serviceImagesData, profileImageData]
```

Comment:
Thoughts on making each of these helpers return an object instead of an array? i.e. 
{isLoading, serviceData, artistData, serviceImagesData, profileImageData}

As the array gets bigger it becomes more difficult for the consumer of the function to retrieve the correct values because they'll need to care about the order of the return values

Reply:
makes sense!